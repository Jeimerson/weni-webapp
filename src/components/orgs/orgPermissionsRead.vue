<template>
  <div class="weni-org-permissions">
    <div class="weni-org-permissions__field">
      <search-user
        v-model="search"
        v-debounce:300ms="onSearch"
        class="weni-org-permissions__input"
        :label="$t('orgs.create.user_search')"
        :placeholder="$t('orgs.create.user_search_description')"
      />
    </div>
    <div class="weni-org-permissions__list">
      <org-role
        v-for="user in permissions"
        disabled
        :can-delete="false"
        :role="user.role"
        :key="user.uuid"
        :email="user.user__email"
        :image-url="user.user__photo"
        :name="isMe(user) ? $t('orgs.you') : user.user__username"
      />
      <infinite-loading ref="infiniteLoading" @infinite="fetchPermissions" />
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import SearchUser from './searchUser';
import OrgRole from './orgRole.vue';
import InfiniteLoading from '../InfiniteLoading';

export default {
  name: 'OrgPermissions',
  components: {
    OrgRole,
    SearchUser,
    InfiniteLoading,
  },
  props: {
    org: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {
      permissions: [],
      search: null,
    };
  },

  methods: {
    ...mapActions([
      'getMembers',
      'addAuthorization',
      'changeAuthorization',
      'removeAuthorization',
    ]),
    async fetchPermissions($state) {
      try {
        const response = await this.getMembers({
          uuid: this.org.uuid,
          page: this.page,
          search: this.search,
        });
        this.page = this.page + 1;
        this.permissions = [...this.permissions, ...response.data.results];
        this.complete = response.data.next == null;
      } catch (e) {
        $state.error();
      } finally {
        if (this.complete) $state.complete();
        else $state.loaded();
      }
    },
    isMe(user) {
      return user.user__username === this.$store.state.Account.profile.username;
    },
    onSearch() {
      this.permissions = [];
      this.page = 0;
      this.complete = false;
      this.$refs.infiniteLoading.reset();
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@weni/unnnic-system/src/assets/scss/unnnic.scss';
.weni-org-permissions {
  display: flex;
  flex-direction: column;
  &__field {
    display: flex;
    margin: 0 0 $unnnic-spacing-stack-md 0;
  }
  &__input {
    flex: 1;
    margin: 0 $unnnic-inline-sm 0 0;
  }
  &__list {
    flex: 1;
    > *:not(:last-child) {
      margin: 0 0 $unnnic-spacing-stack-md 0;
    }
  }

  &__separator {
    border: 1px solid $unnnic-color-neutral-soft;
    margin: $unnnic-spacing-stack-md 0;
  }

  &__button {
    width: 100%;
  }
}
</style>
