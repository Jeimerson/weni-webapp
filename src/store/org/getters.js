export default {
  currentOrg(state) {
    return state.currentOrg;
  },
};
