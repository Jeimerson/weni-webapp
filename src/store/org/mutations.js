export default {
  setCurrentOrg(state, organization) {
    state.currentOrg = organization;
  },
};
