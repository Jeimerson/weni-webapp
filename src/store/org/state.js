export default {
  currentOrgId: null,

  currentOrg: null,
};
