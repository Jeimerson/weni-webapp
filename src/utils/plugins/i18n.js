import Vue from 'vue';
import VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';

Vue.use(VueI18n);

const languages = {
  en: en,
  'pt-br': pt_br,
  'pt-BR': pt_br,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
});

export default i18n;
