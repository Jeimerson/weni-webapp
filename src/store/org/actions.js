import orgs from '../../api/orgs';

export default {
  getOrgs(store, { page = 1, limit = 20 }) {
    const offset = limit * (page - 1);
    return orgs.list(offset, limit);
  },

  createOrg(store, { name, description }) {
    return orgs.createOrg(name, description);
  },

  editOrg(store, { uuid, name, description }) {
    return orgs.editOrg(uuid, name, description);
  },

  deleteOrg(store, { uuid }) {
    return orgs.deleteOrg(uuid);
  },

  getMembers(store, { uuid, page = 1, limit = 20, search }) {
    const offset = limit * (page - 1);
    return orgs.getMembers(uuid, offset, limit, search);
  },

  addAuthorization(store, { orgId, username, role }) {
    return orgs.addAuthorization(orgId, username, role);
  },

  removeAuthorization(store, { orgId, username }) {
    return orgs.removeAuthorization(orgId, username);
  },

  changeAuthorization(store, { orgId, username, role }) {
    return orgs.changeAuthorization(orgId, username, role);
  },

  leaveOrg(store, { orgId, username }) {
    return orgs.leaveOrg(orgId, username);
  },

  setCurrentOrg(
    { commit },
    { name, uuid, inteligence_organization, authorization } = {},
  ) {
    commit('setCurrentOrg', {
      name,
      uuid,
      inteligence_organization,
      authorization,
    });
  },

  clearCurrentOrg({ commit }) {
    commit('setCurrentOrg', null);
  },
};
