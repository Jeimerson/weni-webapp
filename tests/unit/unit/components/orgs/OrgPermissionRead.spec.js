import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import orgPermissionsRead from '@/components/orgs/orgPermissionsRead.vue';
import i18n from '@/utils/plugins/i18n';
import org from '../../../__mocks__/org';

const localVue = createLocalVue();
localVue.use(Vuex);
describe('orgPermissionsRead.vue', () => {
  let wrapper;
  let actions;
  let store;

  beforeEach(() => {
    actions = {
      getMembers: jest.fn(),
      addAuthorization: jest.fn(),
      changeAuthorization: jest.fn(),
      removeAuthorization: jest.fn(),
    };

    store = new Vuex.Store({
      modules: {
        org: {
          actions,
        },
        Account: {
          state: {
            profile: {
              username: 'test',
            },
          },
        },
      },
    });

    wrapper = shallowMount(orgPermissionsRead, {
      localVue,
      store,
      i18n,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        orgRole: true,
        SearchUser: true,
        InfiniteLoading: true,
      },
      propsData: {
        org,
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('test props', async () => {
    expect(wrapper.vm.org).toBe(org);
  });

  it('isMe()', () => {
    const validator = wrapper.vm.isMe({ user__username: 'test' });
    expect(validator).toBeTruthy;
  });

  it('onSearch()', async () => {
    wrapper.vm.$refs.infiniteLoading.reset = jest.fn();
    await wrapper.vm.onSearch();
    expect(wrapper.vm.permissions).toEqual([]);
    expect(wrapper.vm.page).toEqual(0);
    expect(wrapper.vm.complete).toEqual(false);
  });
});
