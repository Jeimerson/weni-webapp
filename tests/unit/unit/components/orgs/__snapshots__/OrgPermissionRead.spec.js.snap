// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orgPermissionsRead.vue renders a snapshot 1`] = `
<div class="weni-org-permissions">
  <div class="weni-org-permissions__field">
    <searchuser-stub label="some specific text" placeholder="some specific text" class="weni-org-permissions__input"></searchuser-stub>
  </div>
  <div class="weni-org-permissions__list">
    <infiniteloading-stub></infiniteloading-stub>
  </div>
</div>
`;
